### ISSUE TITLE
Date of birth formating for HPP form datepicker

### ISSUE THREAD

**Issue by danielozano** [Adyen/adyen-magento2#211]
**Magento version**: 2.1.7
**Plugin version**: 2.1.2
**Description**
When using the M2 datepicker inside the HPP payment method component, the module doesn't takes into account the configured date time format for the current locale.
The module implements the datepicker with the following options:
```json
{
    "options": {
         "showOn": "both", 
         "changeYear": true, 
         "changeMonth": true, 
         "yearRange": "-99:-1",
         "defaultDate": "-20y"
    }
}
```
If you don't specify the option dateFormat, it picks the default (US) set inside the magento js library:
```javascript
    var defaults = {
        dateFormat: 'mm\/dd\/yyyy',
        showsTime: false,
        timeFormat: null,
        buttonImage: null,
        buttonImageOnly: null,
        buttonText: $t('Select Date')
    };
```
https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Ui/view/base/web/js/lib/knockout/bindings/datepicker.js#L17-L24

Steps to reproduce:
- Install Magento 2 + Adyen Module
- Set store locale for any EU Country (for example nl_NL, you need the date format = 'd/m/yyyy')
- Inside the HPP Afterpay input: Date Of Birth select any date with the datepicker.
You will see the date with the following format inside the input: 12/19/1997 instead of 19/12/1997 expected.

I don't know if it's an expected behaviour, but it's really odd for me. If you want I can prepare a PR with my solution.

Thank you,
regards.

**Comment by rikterbeek** [comment]
Hi @danielozano,

Magento core should be responsible for this. Looking into this issue for the front-end it should be fixed in 2.2.1. see https://github.com/magento/magento2/issues/6350 for more details.

Regards,
Rik
Adyen

**Comment by danielozano** [comment]
Hi @rikterbeek, well understood. We'll treat it as a custom implementation then.
Thanks for the fast response! 

PD: For me this issue is solved, if you decide to close it!
